main/lsp/ExtractVariable.h

```diff
@@ -0,0 +1,17 @@
+#ifndef SORBET_EXTRACT_VARIABLE_H
+#define SORBET_EXTRACT_VARIABLE_H
+
+#include "main/lsp/LSPConfiguration.h"
+#include "main/lsp/LSPTypechecker.h"
+#include "main/lsp/json_types.h"
+
+namespace sorbet::realmain::lsp {
+
+std::vector<std::unique_ptr<TextDocumentEdit>> getExtractVariableEdits(LSPTypecheckerDelegate &typechecker,
```

Comment:
Slight preference for not polluting the top-level `lsp` namespace with one-off functions. You can see how `sig_finder.{h,cc}` do this, where the main public method is a static method inside a class, so you'd have something like


```
VariableExtractor::getEdits
```

Reply:
Do we need to take ownership of the selectionRange? It's unexpected for me that this method would require mutating the enclosing `LSPTask::params`. It means that no one else could use the `params->range` in code_action.cc unless they duplicate the `Range` first. But this method isn't mutating the selectionRange, so it could be taken like `const Range& selectionRange`.

Also in general, we should avoid using LSP specification types for method arguments outside of helper functions, especially when there are suitable core Sorbet data structures that model them.

Also, it looks like the one call to `getExtractVariableEdits` passes both `move(params->range)` and also `loc`. But `loc` was already created by converting `params->range` to a `core::Loc`, so it seems like it takes the same argument twice in different forms